azurerm/internal/services/eventhub/data_source_eventhub_namespace_authorization_rule.go

```diff
@@ -0,0 +1,79 @@
+package eventhub
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceEventHubNamespaceAuthorizationRule() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceEventHubNamespaceAuthorizationRuleRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: azure.EventHubAuthorizationRuleSchemaFrom(map[string]*schema.Schema{
```

Comment:
THis puts the properties in as non computed, we need to take that schema and place it in here inline as computed istead.